### PR TITLE
Investigating #84

### LINKED ISSUE #84: Possible deadlock when running Python 3.11+ tests
Running the tests for Python 3.11 and 3.12 on GHA is currently cancelled after six hours of runtime.
Maybe we have (again) a deadlock here.

Example runs:
* Python 3.11: https://github.com/zopefoundation/zope.sqlalchemy/actions/runs/10870806212/job/30163863945
* Python 3.12: https://github.com/zopefoundation/zope.sqlalchemy/actions/runs/9986832358/job/27623911631?pr=83

### PR DESCRIPTION
**Pull request by icemac**
Fixes #84.

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -4,8 +4,9 @@ Changes
 3.2 (unreleased)
 ----------------
 
-- Nothing changed yet.
-
+- SQLAlchemy's versions 2.0.32 up to 2.0.35 run into dead locks when running
+  the tests on Python 3.11+, so excluding them from the list of supported
+  versions.
```
Comment (dataflake) [inline CHANGES.rst:9]:
```suggestion
  versions.
  (`#84 <https://github.com/zopefoundation/zope.sqlalchemy/issues/84>`_)
```

File: buildout.cfg [hunk 2]
```diff
@@ -1,6 +1,7 @@
 [buildout]
 develop = .
 parts = test scripts
+show-picked-versions = true
```
Comment (dataflake) [inline buildout.cfg:4]: Is this a leftover diagnostic configuration?

**Review by dataflake — CHANGES_REQUESTED** [review]

File: buildout.cfg (hunk 2 above)
Comment (icemac) [reply buildout.cfg:4]: I intentionally left it there for future diagnosis: `pip` shows the versions it has installed buildout does not by default and we do not use version pins here.

**Comment by icemac** [comment]
@dataflake Please re-review.

**Review by dataflake — CHANGES_REQUESTED** [review]
👍

**Review by dataflake — APPROVED** [review]

**Pull request merged**